### PR TITLE
Revamp webhook spec

### PR DESCRIPTION
**Pull request by carmenlau**
refs #1041

### CODE REVIEW

File: docs/specs/webhook.md [hunk 1]
```diff
@@ -142,77 +124,71 @@ If any handler fails the operation, the operation is failed. The operation fails
...
+- [identity.created.admin_api_add_identity](#identitycreateduser_add_identity-identitycreatedadmin_api_add_identity)
+- [identity.deleted.user_remove_identity](#identitydeleteduser_remove_identity-identitydeletedadmin_api_remove_identity)
+- [identity.deleted.admin_api_remove_identity](#identitydeleteduser_remove_identity-identitydeletedadmin_api_remove_identity)
+- [identity.updated.user_update_identity](#identityupdateduser_update_identity)
+- [session.created.user_signup](#sessioncreateduser_signup-sessioncreateduser_login)
+- [session.created.user_login](#sessioncreateduser_signup-sessioncreateduser_login)
+- [session.deleted.user_revoke_session](#sessiondeleteduser_revoke_session-sessiondeleteduser_logout-sessiondeletedadmin_api_revoke_session)
+- [session.deleted.user_logout](#sessiondeleteduser_revoke_session-sessiondeleteduser_logout-sessiondeletedadmin_api_revoke_session)
+- [session.deleted.admin_api_revoke_session](#sessiondeleteduser_revoke_session-sessiondeleteduser_logout-sessiondeletedadmin_api_revoke_session)
+- [user.promoted.user_promote](#userpromoteduser_promote)
```
Comment (louischan-oursky) [inline docs/specs/webhook.md:178]: Grammatically the verb "promote" is transitive verb so if we put it into full sentence, the sentence would be "The user promoted themselves" or "The user was promoted"

But it looks a bit ambiguous if we define the event name as `user.promoted.user_promote_user`.
Or it is not consistent if we use passive voice only for this event. Would `user.promoted.user_promote_themselves` a good name?

Comment (carmenlau) [reply docs/specs/webhook.md:178]: IMO `user.promoted.user_promote_themselves` is better than `user.promoted.user_promote_user`, let me change it!

File: docs/specs/webhook.md [hunk 2]
```diff
@@ -66,7 +66,8 @@ All fields are guaranteed that only backward-compatible changes would be made.
 
 - `timestamp`: signed 64-bit UNIX timestamp of when this event is generated. Retried deliveries do not affect this field.
 - `user_id`: The ID of the user associated with the event. It may be absent. For example, the user has not authenticated yet.
-- `locale`: User locale, derive based on user's preferred languages and app's supported languages.
+- `preferred_languages`: User preferred languages which are inferred from the request. Return values of `ui_locales` query if it is provided in auth ui, otherwise return languages in `Accept-Language` request header.
+- `resolved_language`: User locale which is derived based on user's preferred languages and app's languages config.
```
Comment (louischan-oursky) [inline docs/specs/webhook.md:70]: Rename it to `language`

**Comment by carmenlau** [comment]
Updated the spec

**Comment by louischan-oursky** [comment]
Looks good to me. Can I merge it?

**Comment by carmenlau** [comment]
Yes, we agreed the events in the meeting, I think we can merge it first. @chpapa You can take a look at this PR when you have time, in case there are further comments, we can address them in another PR.

Thanks all :)

**Pull request merged**